src/System.Device.Gpio/System/Device/Pwm/Channels/UnixPwmChannel.Linux.cs

```diff
@@ -0,0 +1,201 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Threading;
+
+namespace System.Device.Pwm
+{
+    /// <summary>
+    /// Represents a PWM channel running on Unix.
+    /// </summary>
+    internal class UnixPwmChannel : PwmChannel
+    {
+        private readonly int _chip;
+        private readonly int _channel;
+        private double _frequency;
+        private double _dutyCyclePercentage;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UnixPwmChannel"/> class.
+        /// </summary>
+        /// <param name="chip">The PWM chip number.</param>
+        /// <param name="channel">The PWM channel number.</param>
+        /// <param name="frequency">The frequency in hertz.</param>
+        /// <param name="dutyCyclePercentage">The duty cycle percentage represented as a value between 0.0 and 1.0.</param>
+        /// <param name="polarity">The polarity of the PWM signal.</param>
+        public UnixPwmChannel(
+            int chip,
+            int channel,
+            double frequency,
+            double dutyCyclePercentage = 0.5,
+            PwmPolarity polarity = PwmPolarity.ActiveHigh)
+        {
+            _chip = chip;
+            _channel = channel;
+            Validate();
+            Open();
+            Thread.Sleep(100);  // TODO: Need a better solution to delay for available file.
+
+            SetFrequency(frequency);
+            DutyCyclePercentage = dutyCyclePercentage;
+
+            // NOTE: There appears to be a lower-level bug that won't allow the
+            // polarity to be changed until a valid period has been written.
+            SetPolarity(polarity);
+        }
+
+        /// <summary>
+        /// The frequency in hertz.
+        /// </summary>
+        public override double Frequency =>_frequency;
+
+        /// <summary>
+        /// The duty cycle percentage represented as a value between 0.0 and 1.0.
+        /// </summary>
+        public override double DutyCyclePercentage
+        {
+            get
+            {
+                return _dutyCyclePercentage;
+            }
+            set
+            {
+                SetDutyCyclePercentage(value);
+            }
+        }
+
+        /// <summary>
+        /// Gets the frequency period in nanoseconds.
+        /// </summary>
+        /// <param name="frequency">The frequency in hertz.</param>
+        /// <returns>The frequency period in nanoseconds.</returns>
+        private static int GetPeriodInNanoSeconds(double frequency)
+        {
+            // In Linux, the period needs to be a whole number and can't have a decimal point.
+            return (int)((1.0 / frequency) * 1_000_000_000);
+        }
+
+        // Helpers for file paths.
+        private string GetChipPath() => $"/sys/class/pwm/pwmchip{_chip}";
```

Comment:
Should we just make these 3 private fields instead of methods? I don't really see the point of having them be methods.

Reply:
Yes, this was updated in last commit